compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -485,11 +485,22 @@ private BlendedManifest createBlendedManifest(PackageContext rootPackageContext,
 
     private ResolutionOptions getResolutionOptions(PackageContext rootPackageContext,
                                                    CompilationOptions compilationOptions) {
+        boolean sticky = getSticky(rootPackageContext);
+        PackageLockingMode packageLockingMode;
+        SemanticVersion prevDistributionVersion = rootPackageContext.dependencyManifest().distributionVersion();
+        SemanticVersion currentDistributionVersion = SemanticVersion.from(RepoUtils.getBallerinaShortVersion());
+        if (!sticky && (null == prevDistributionVersion
+                || currentDistributionVersion.greaterThan(prevDistributionVersion))) {
```

Comment:
We must issue a warning diagnostic informing the user about the automatic update policy change.

Reply:
Should the warning diagnostic shown only if there is a distribution version increment (i.e. only if the if condition in L492 met)?

Reply:
Only if the condition is met. 
E.g:
`WARNING: Detected packages built with an older SL Update distribution. Dependencies will be updated to the latest versions compatible with the current distribution.
`